File: JavaScript/models/Clicker.js
import Cordyceps from './Cordyceps.js';

class Clicker extends Cordyceps {
    //constructor
    constructor(origin) {
        super(origin);
        this._sound = "¡Click!";
    }

    //metodo con retorno
    behavior() {
        return "En esta fase el hongo ha dejado ciego al infectado, usa ecolocalización para cazar.\nConsejo: no lo enfrentes a mano limpia, evita el enfrentamiento.";
    }

    //metodo sin retorno
    emitSound() {
        console.log("Sonido emitido: " + this._sound);
    }

    // funcion con retorno
    getSound() {
        return this._sound;
    }

    //funcion sin retorno con parametro
    changeSound(newSound) {
        this._sound = newSound;
        console.log("El chasqueador ha hecho un sonido distinto: " + this._sound);
    }
}

export default Clicker;

File: GO/models/clicker.go
package models

import "fmt"

type Clicker struct {
    BaseCordyceps
    sound string // atributo privado
}

// constructor
func NewClicker(origin string) *Clicker {
    return &Clicker{
        BaseCordyceps: BaseCordyceps{origin: origin},
        sound:         "¡Click!",
    }
}

// Método con retorno
func (c *Clicker) Behavior() string {
    return "En esta fase el hongo ha dejado ciego al infectado, usa ecolocalización para cazar.\nConsejo: no lo enfrentes a mano limpia, evita el enfrentamiento."
}

// Método sin retorno
func (c *Clicker) EmitSound() {
    fmt.Println("Sonido emitido: " + c.sound)
}

// Función con retorno
func (c *Clicker) GetSound() string {
    return c.sound
}

// Función sin retorno con parámetro
func (c *Clicker) ChangeSound(newSound string) {
    c.sound = newSound
    fmt.Println("El chasqueador ha hecho un sonido distinto: " + c.sound)
}
